gramps/gen/filters/rules/person/_regexpname.py

```diff
@@ -53,16 +62,16 @@ class RegExpName(Rule):
     category = _("General filters")
     allow_regex = True
 
-    def apply(self, db, person):
-        for name in [person.get_primary_name()] + person.get_alternate_names():
+    def apply_to_one(self, db: Database, person: Person) -> bool:
+        for name in [person.primary_name] + person.alternate_names:
             for field in [
                 name.first_name,
-                name.get_surname(),
                 name.suffix,
                 name.title,
                 name.nick,
                 name.famnick,
                 name.call,
+                name.get_surname(),
```

Comment:
Is there a reason for changing the order in which fields are tested

Reply:
Yes, it is more expensive, so do the others first.